src/webgpu/shader/execution/expression/call/builtin/texture_utils.ts

```diff
@@ -412,26 +421,49 @@ export function softwareRasterize<T extends Dimensionality>(
   options: TextureTestOptions
 ) {
   const [width, height] = targetSize;
-  const { ddx = 1, ddy = 1, ddxStart = 0, ddyStart = 0 } = options;
+  const { ddx = 1, ddy = 1, uvwStart = [0, 0] } = options;
   const format = 'rgba32float';
 
+  const textureSize = reifyExtent3D(texture.descriptor.size);
+  const screenSpaceDDX = (ddx * width) / textureSize.width;
+  const screenSpaceDDY = (ddy * height) / textureSize.height;
+
   const rep = kTexelRepresentationInfo[format];
 
   const expData = new Float32Array(width * height * 4);
   for (let y = 0; y < height; ++y) {
     const fragY = height - y - 1 + 0.5;
     for (let x = 0; x < width; ++x) {
       const fragX = x + 0.5;
-      const coords = [(fragX / width) * ddx - ddxStart, (fragY / height) * ddy - ddyStart] as T;
+      // This code calculates the same value that will be passed to
+      // `textureSample` in the fragment shader for a given frag coord. That
+      // shader renders a clip space quad and includes a inter-stage "uv"
+      // coordinates that start with a unit quad (0,0) to (1,1) and is
+      // multiplied by ddx,ddy and as added in uStart and vStart
+      //
+      // uv = unitQuad * vec2(ddx, ddy) + vec2(vStart, uStart);
+      //
+      // softwareTextureRead<T> simulates a single call to `textureSample` so
+      // here we're computing the `uv` value that will be passed for a
+      // particular fragment coordinate. fragX / width, fragY / height provides
+      // the unitQuad value.
+      //
+      // ddx and ddy in this case are the derivative values we want to test. We
+      // pass those into the softwareTextureRead<T> as they would normally be
+      // derived from the change in coord.
+      const coords = [
+        (fragX / width) * screenSpaceDDX + uvwStart[0],
+        (fragY / height) * screenSpaceDDY + uvwStart[1],
```

Comment:
It was a bug fix. They should both be added. The reason it worked before (I think) is because I was adding an integer start it didn't affect the results because `'repeat'`. I changed the test start to a non-integer